### ISSUE TITLE
Issue while fetching offerid for a successful transaction with manage offer operation

### ISSUE THREAD

**Issue by swetakedia** [stellar/java-stellar-sdk#47]
I am getting nullpointer exception while I am trying to fetch the offer id from the SubmitTransactionResponse Object.

```
ManageOfferOperation manageOfferOp = new ManageOfferOperation.Builder(sendingAsset, receivingAsset, "12", "0.5").setOfferId(0).setSourceAccount(issuingAccount).build();
		
		Transaction manageOffer = new Transaction.Builder(source)
		  .addOperation(manageOfferOp)
		  .build();
		manageOffer.sign(issuingKeys);
		
		System.out.println("Transaction xdr " + manageOffer.toEnvelopeXdrBase64());
		SubmitTransactionResponse  subtranresponse= server.submitTransaction(manageOffer);
			
		System.out.println("subtranresponse.isSuccess(): "+ subtranresponse.isSuccess());
		if(subtranresponse.isSuccess())
		{
			Long offerid=subtranresponse.getOfferIdFromResult(0);
			if(null!=subtranresponse.getOfferIdFromResult(0)){
				System.out.println("offerid: "+ offerid);
				System.out.println("subtranresponse.getEnvelopeXdr(): "+ subtranresponse.getEnvelopeXdr());
				System.out.println("subtranresponse.getResultXdr(): "+ subtranresponse.getResultXdr());
			System.out.println("Offer submitted successfully" + subtranresponse.getOfferIdFromResult(0));
			System.out.println("Get hash:"+ subtranresponse.getHash() + "Ledger: "+ subtranresponse.getLedger().toString()) ;
			}
		}

```
I am getting nullpointerexception in 	`Long offerid=subtranresponse.getOfferIdFromResult(0);` line

This behavior is random. When I retry submitting the txn with same code and input param, I get the success response with proper offer-id.

However value of `subtranresponse.isSuccess()` is always true.

Tried using local horizon server as well as Stellar's horizon.

**Comment by bartekn** [comment]
I'm not able to reproduce it. Could you send the value of `subtranresponse.getResultXdr()` for the execution that results in `java.lang.NullPointerException` being thrown (you can just print it before the line that's causing `java.lang.NullPointerException`)?

**Comment by swetakedia** [comment]
PFB the required data:

Transaction xdr AAAAAP0uDCJGzWofMAUkG/F3YCPctgLkeP3VTr3P1mHKznAHAAAAZAA5klgAAABKAAAAAAAAAAAAAAABAAAAAQAAAAD9LgwiRs1qHzAFJBvxd2Aj3LYC5Hj91U69z9Zhys5wBwAAAAMAAAABRVVSAAAAAAD9LgwiRs1qHzAFJBvxd2Aj3LYC5Hj91U69z9Zhys5wBwAAAAFVU0QAAAAAAOw4Vbv3zJ23hiG1bjU7M/AOuVNHKnREERoitIr1Zj/ZAAAAAAOThwAAAAACAAAAAQAAAAAAAAAAAAAAAAAAAAHKznAHAAAAQGK49TjFF8thkJA7x1ZYXy1V0KqYeQllcM0l1coVT5QlBuzamttVyUr4Ggu69jdHtHQhQ6Fjb19jHPWuuNI3CAg=
subtranresponse.isSuccess(): true
Inside success: 
java.lang.NullPointerException
       at org.stellar.sdk.responses.SubmitTransactionResponse.getOfferIdFromResult(SubmitTransactionResponse.java:98)
       at com.stellar.Offers.manageOffer(Offers.java:102)
       at com.stellar.Offers.main(Offers.java:30)

**Comment by jedmccaleb** [comment]
But what is the result xdr? Likely the offer was taken when you placed it so there is no offer left

**Comment by swetakedia** [comment]
Transaction xdr AAAAAP0uDCJGzWofMAUkG/F3YCPctgLkeP3VTr3P1mHKznAHAAAAZAA5klgAAABLAAAAAAAAAAAAAAABAAAAAQAAAAD9LgwiRs1qHzAFJBvxd2Aj3LYC5Hj91U69z9Zhys5wBwAAAAMAAAABRVVSAAAAAAD9LgwiRs1qHzAFJBvxd2Aj3LYC5Hj91U69z9Zhys5wBwAAAAFVU0QAAAAAAOw4Vbv3zJ23hiG1bjU7M/AOuVNHKnREERoitIr1Zj/ZAAAAAAOThwAAAAACAAAAAQAAAAAAAAAAAAAAAAAAAAHKznAHAAAAQEe1jyNCeK3TckPuuKeWIICf6nvz2zBZ8mbbUamWLnOFMMqvQPTllOe9DIdloNxaixgle9zi2F+yyOhLzpNhkAg=
subtranresponse.isSuccess(): true
Inside success: 
subtranresponse.getEnvelopeXdr(): AAAAAP0uDCJGzWofMAUkG/F3YCPctgLkeP3VTr3P1mHKznAHAAAAZAA5klgAAABLAAAAAAAAAAAAAAABAAAAAQAAAAD9LgwiRs1qHzAFJBvxd2Aj3LYC5Hj91U69z9Zhys5wBwAAAAMAAAABRVVSAAAAAAD9LgwiRs1qHzAFJBvxd2Aj3LYC5Hj91U69z9Zhys5wBwAAAAFVU0QAAAAAAOw4Vbv3zJ23hiG1bjU7M/AOuVNHKnREERoitIr1Zj/ZAAAAAAOThwAAAAACAAAAAQAAAAAAAAAAAAAAAAAAAAHKznAHAAAAQEe1jyNCeK3TckPuuKeWIICf6nvz2zBZ8mbbUamWLnOFMMqvQPTllOe9DIdloNxaixgle9zi2F+yyOhLzpNhkAg=
subtranresponse.getResultXdr(): AAAAAAAAAGQAAAAAAAAAAQAAAAAAAAADAAAAAAAAAAEAAAAA7DhVu/fMnbeGIbVuNTsz8A65U0cqdEQRGiK0ivVmP9kAAAAAAAASJQAAAAFVU0QAAAAAAOw4Vbv3zJ23hiG1bjU7M/AOuVNHKnREERoitIr1Zj/ZAAAAAAcnDgAAAAABRVVSAAAAAAD9LgwiRs1qHzAFJBvxd2Aj3LYC5Hj91U69z9Zhys5wBwAAAAADk4cAAAAAAgAAAAA=
java.lang.NullPointerException
	at org.stellar.sdk.responses.SubmitTransactionResponse.getOfferIdFromResult(SubmitTransactionResponse.java:98)
	at com.stellar.Offers.manageOffer(Offers.java:101)
	at com.stellar.Offers.main(Offers.java:26)

**Comment by bartekn** [comment]
As Jed [mentioned](https://github.com/stellar/java-stellar-sdk/issues/47#issuecomment-328768530) you won't get offer id in `result_xdr` if the offer was taken. This is by design.

**Comment by bartekn** [comment]
We should change `getOfferIdFromResult` to return `null` in this case.

**Comment by bartekn** [comment]
Fixed in [0.1.13](https://github.com/stellar/java-stellar-sdk/releases/tag/0.1.13).
